### DISCUSSION TITLE
New Archaeologist Type

### DISCUSSION THREAD

**Discussion by adamgall** [sarcophagus-org/sarcophagus-v2-specs#4] [Ideas]
# What's the problem

Running an archaeologist is no small feat.

One point of friction is that an archaeologist needs to write data to _both_ Ethereum Mainnet _and_ Arweave Mainnet. This requires the archaeologist to have economic exposure to both of those blockchains/cryptocurrencies. This can be a significant bottleneck to individuals who might not own both cryptocurrencies, and haven't figured out their own risk profiles for both ecosystems.

From a technical side, having a single archaeologist be responsible for too many things increases the risk of total failure for that archaeologist, and can also induce a false sense of security if one of their responsibilities fails, but the other works.

# Let's fix it

We can solve these issues, and simplify the jobs of archaeologists, by splitting the current responsibilities of the single archaeologist into two different archaeologist services.

Now, individuals and institutions can choose to run either service, or both! People may have their own reasons for wanting to run one, or the other, or both archaeologist services.

One service will write _only_ to the Arweave chain. This service won't need to hold ETH for gas fees, and will only need to be online at the moment of sarcophagus creation, so it has no long-term availability requirements.

The other service will write _only_ to the Ethereum chain. This service won't need to hold AR for gas fees, and will need to be long running, so that it can perform its duties at the time of unwrapping (dead man's switch triggering).

TODO: need some good names for these services

**Comment by sethhrbek** [comment]
What if the "AR" is capitalized in this version -- ARchaeologist

**Reply by adamgall** [reply]
this is literally blowing my mind right now

**Reply by USCMig** [reply]
Could Bundlr be leveraged to simplify the payment and/or need to have At tokens for the ARchaeologist? Just stumbled across this and could make the maintenance aspect of running a node easier if they just have to worry about Carrying ETH... similar to the zapper conversations that we had, but this is specifically designed for interactions with Arweave. https://docs.bundlr.network/

**Reply by sethhrbek** [reply]
Bundlr looks pretty straightforward. The ARch could potentially use Bundlr, but I would argue Bundlr should just replace the ARch in the sarcophagus creation process and the embalmer can use it directly (if Bundlr is feasible).

My guess is the Bundlr UX from the embalmer's point of view would look pretty similar to the ARch UX (use 3rd party to upload file to arweave to avoid needing AR). 

Need to determine Bundlr feasibility (is their mainnet live yet? if not, when?), and also understand their security model + weigh their chance of longevity. It would save some dev time (that we could devote elsewhere on the project) if v2 used Bundlr, so def worth a look IMO. It would also remove the need for the sarcophagus community to incentivize / recruit ARchs, and offload that incentive work to Bundlr (which they incentivize with their token)

**Reply by sethhrbek** [reply]
> Bundlr Network comprises nodes (known as bundlers), which are used to bundle multiple layer 2 transactions into a single Arweave (layer 1) transaction. Each will periodically submit these bundles to the main Arweave network. Once a user submits a valid transaction to a bundler, they receive a receipt which acts as a financial commitment that the data will be seeded on layer 1 within an n number of blocks (where n is defined in the receipt). Seeding is handled automatically by Bundlr.
> 
> Validators ensure data is uploaded correctly, by the block cutoff period defined in the receipt.

**Reply by sethhrbek** [reply]
For Bundlr, Wonder what the avg delay will be from when a TX is submitted to a Bundlr until when it is finalized on arweave ------  If there's a large delay and the embalmer can't validate the data on layer 1, it would have to trust the node to do its job (which may be a non-starter).

This shouldn't be an issue with the ARch, b/c those TXs would be submitted on-demand (like how it works in v1)

**Comment by adamgall** [comment]
Update

This idea may be adding too much system-level complexity, and end-user transactional cost, to be worth it.
